src/tests/MultiselectProblem.test.tsx

```diff
@@ -0,0 +1,185 @@
+import { fireEvent, render, screen, act } from '@testing-library/react'
+import { MultiselectProblem } from '../components/MultiselectProblem'
+import '@testing-library/jest-dom'
+
+test('MultiselectProblem renders', async () => {
+  render(
+    <MultiselectProblem
+    solutionOptions={'["Option 1", "Option 2", "Option 3"]'}
+    solvedCallback={() => {}}
+    exhaustedCallback={() => {}}
+    allowedRetryCallback={() => {}}
+    content={'<p>Question Content</p>'}
+    correctResponse={''}
+    encourageResponse={''}
+    retryLimit={0}
+    solution={'[]'}
+    buttonText={'Check'}
+    />
+  )
+  screen.getByText('Question Content')
+  screen.getByText('Option 1')
+  screen.getByText('Option 2')
+  screen.getByText('Option 3')
+  expect(screen.getByRole('button')).toHaveTextContent('Check')
+})
+
+test('Multiselect shows message if user does not select an option', async () => {
+  render(
+    <MultiselectProblem
+    solutionOptions={'["Option 1", "Option 2", "Option 3"]'}
+    solvedCallback={() => {}}
+    exhaustedCallback={() => {}}
+    allowedRetryCallback={() => {}}
+    content={'<p>Problem text</p>'}
+    correctResponse={''}
+    encourageResponse={''}
+    retryLimit={0}
+    solution={'[]'}
+    buttonText={'Check'}
+    />
+  )
+
+  await act(async () => {
+    screen.getByRole('button').click()
+  })
+  await screen.findByText('Please select an answer')
+})
+
+test('Multiselect shows correct response, invokes callback, and disables self on check with match', async () => {
+  const solvedHandler = jest.fn()
+  const exhaustedHandler = jest.fn()
+  const allowedRetryHandler = jest.fn()
+
+  render(
+    <MultiselectProblem
+    solutionOptions={'["Option 1", "Option 2", "Option 3"]'}
+    solvedCallback={solvedHandler}
+    exhaustedCallback={exhaustedHandler}
+    allowedRetryCallback={allowedRetryHandler}
+    content={'<p>Problem text</p>'}
+    correctResponse={'<p>Great job!</p>'}
+    encourageResponse={''}
+    retryLimit={0}
+    solution={'["Option 2", "Option 3"]'}
+    buttonText={'Check'}
+    />
+  )
+
+  await act(async () => {
+    fireEvent.click(screen.getByLabelText('Option 2'))
+    fireEvent.click(screen.getByLabelText('Option 3'))
```

Comment:
You're currently missing coverage that "unclicks" work (the `newSet.delete(e.target.value)` line highlighted by codecov). You might want to create a dedicated test case for this, but you could also just click and then unclick `Option 1` here if you wanted to :man_shrugging: .